.cirrus.yml

```diff
@@ -30,13 +31,21 @@ nix_test_task:
 task:
   auto_cancellation: true
 
-  env:
-    matrix:
-      PY_VER: "3.6"
-      PY_VER: "3.7"
+  matrix:
+    # not a pull request, run full suite
+    - only_if: $CIRRUS_PR == ""
+      env:
+        matrix:
+          PY_VER: "3.6"
+          PY_VER: "3.7"
+          PY_VER: "3.8"
+    # is a pull request, only run python 3.8
+    - only_if: $CIRRUS_PR != ""
+      env:
+        PY_VER: "3.8"
 
   matrix:
```

Comment:
wow, I'm surprised that this worked since in normal YAML, it would've been overwritten. good to know!